guides/05_tabular_data_science_and_plots/creating_a_realtime_dashboard_from_google_sheets.md

```diff
@@ -0,0 +1,147 @@
+
+Building the dashboard itself with just be 9 lines of Python code using Gradio, and our final dashboard will look like this:
+
+<gradio-app space="gradio/line-plot"></gradio-app>
+
+**Prerequisites**: This Guide uses [Gradio Blocks](../01_getting_started/01_quickstart.md#blocks-more-flexibility-and-control), so make your are familiar with the Blocks class. 
+
+The process is a little different depending on if you are working with a publicly accessible or a private Google Sheet. We'll cover both, so let's get started!
+
+## Public Google Sheets
+
+If you want to build a dashboard from a Google Sheet that is public, then loading the data is very easy, thanks to the `pandas` library.
+
+1. Get the URL of the Google Sheets that we want to use. To do this, simply go to the Google Sheets, click on the "Share" button in the top-right corner, and then click on the "Get shareable link" button. This will give you a URL that looks something like this:
+
+```html
+https://docs.google.com/spreadsheets/d/1UoKzzRzOCt-FXLLqDKLbryEKEgllGAQUEJ5qtmmQwpU/edit#gid=0
+```
+
+2. Now, let's modify this URL and then use it to read the data from the Google Sheets into a Pandas DataFrame. (In the code below, replace the `URL` variable with the URL of your public Google Sheet):
+
+```py
+import pandas as pd
+
+URL = "https://docs.google.com/spreadsheets/d/1UoKzzRzOCt-FXLLqDKLbryEKEgllGAQUEJ5qtmmQwpU/edit#gid=0"
+csv_url = URL.replace('/edit#gid=', '/export?format=csv&gid=')
+
+def get_data():
+    return pd.read_csv(csv_url)
+```
+
+3. We've made the data query a function, which means that it's easy to display it real-time using the the `gr.DataFrame` component, or plot it real-time using the `gr.LinePlot` component (of course, depending on the data, a different plot may be appropriate). To do this, we just pass the function into the respective components, and set the `every` parameter based on how frequently (in seconds) we would like the component to refresh. Here's the Gradio code:
+
+```py
+import gradio as gr
+
+with gr.Blocks() as demo:
+    gr.Markdown("# 📈 Real-Time Line Plot")
+    with gr.Row():
+        with gr.Column():
+            gr.DataFrame(get_data, every=5)
+        with gr.Column():
+            gr.LinePlot(get_data, every=5, x="Date", y="Sales", y_title="Sales ($ millions)", overlay_point=True, width=500, height=500)
+
+demo.queue().launch()  # Run the demo with queuing enabled
+```
+ 
+And that's it! You have a Dashboard that refreshes every 5 seconds, pulling the data from your Google Sheet.
+
+## Private Google Sheets
+
+For private Google Sheets, the process requires a little more work, but not that much! The key difference is that now, we must authenticate ourselves so to authorize access to the private Google Sheets.
```

Comment:
typo: 
> ourselves so to authorize

ourselves ~so~ to authorize

Reply:
It would be cool to try creating first a numbered list of the headers as steps. to see if they would show up on search. something like 

> To build a dashboard from a private Google sheet: 
> 1. Set up credentials and authentication
> 2. Query the data 
> 3. Display the data

It's good to keep in mind what users will search for

Reply:
Fixed the typo. I feel like the numbered list disturbs the flow of the guide